src/video/x11/SDL_x11video.c

```diff
@@ -178,10 +184,22 @@ X11_CreateDevice(int devindex)
         return NULL;
     }
     device->driverdata = data;
+    device->wakeup_lock = SDL_CreateMutex();
 
     data->global_mouse_changed = SDL_TRUE;
 
     data->display = x11_display;
+    data->request_display = X11_XOpenDisplay(display);
```

Comment:
Why open a separate `request_display`?

Reply:
This is needed based on my knowledge of X11 and also my past experience. If you wait on a socket for an event (the display is the socket) you shouldn't write on the same socket (display) from another thread. Expert people told me this is basically UB and can give weird problems even if may appear to work at first.

For this reason we keep it safe and far from UB by sending the wake-up event in a dedicate display connection. This way it works reliably AFAIK.